addons/account/models/account_account.py

```diff
@@ -685,8 +735,9 @@ def copy_data(self, default=None):
         cache_map = defaultdict(set)
         for account, vals in zip(self, vals_list):
             if 'code' not in default:
-                company = default.get('company_id', account.company_id)
-                company = company if isinstance(company, models.BaseModel) else account.env['res.company'].browse(company)
+                companies = default.get('company_ids', account.company_ids)
+                companies = companies if isinstance(companies, models.BaseModel) else account.env['res.company'].browse(companies)
+                company = self.env.company if self.env.company in companies else (companies & self.env.companies)[:1]
                 cache = cache_map[company.id]
                 vals['code'] = account._search_new_account_code(account.code, company, cache)
```

Comment:
Note that this doesn't guarantee that the code will be available in all companies